### PR TITLE
[IA-3035] Add GoogleComputeService.setInstanceTags

### PR DESCRIPTION
**Pull request by rtitle**
Needed for https://broadworkbench.atlassian.net/browse/IA-3035

**PR checklist**
- [ ] For each library you've modified here, decide whether it requires a major, minor, or no version bump. (Click [here](/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md) for guidance)

If you're doing a **major** or **minor** version bump to any libraries:
- [ ] Bump the version in `project/Settings.scala` `createVersion()`
- [ ] Update `CHANGELOG.md` for those libraries
- [ ] I promise I used `@deprecated` instead of deleting code where possible

In all cases:
- [ ] Replace the appropriate version hashes in `README.md` and the `CHANGELOG.md` for any libs you changed with `TRAVIS-REPLACE-ME` to auto-update the version string
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green (CI _and_ coverage tests)
- [ ] Squash commits and **merge to develop**
- [ ] Delete branch after merge

### CODE REVIEW

File: google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleComputeInterpreter.scala [hunk 1]
```diff
@@ -297,6 +297,15 @@ private[google2] class GoogleComputeInterpreter[F[_]: Parallel: StructuredLogger
       s"com.google.cloud.compute.v1.SubnetworksClient.insertSubnetwork(${project.value}, ${region.value}, ${subnetwork.getName})"
     )
 
+  override def setInstanceTags(project: GoogleProject, zone: ZoneName, instanceName: InstanceName, tags: Tags)(implicit
+    ev: Ask[F, TraceId]
+  ): F[Operation] =
+    retryF(
+      F.delay(instanceClient.setTags(project.value, zone.value, instanceName.value, tags)),
+      s"com.google.compute.v1.InstancesClient.setTags(${project.value}, ${zone.value}, ${instanceName.value}, [${tags.getItemsList.asScala
```
Comment (Qi77Qi) [inline google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleComputeInterpreter.scala:306]: this is setting tag on the instance, I thought AoU needs to set tags on the network of the instance?

Comment (rtitle) [reply google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleComputeInterpreter.scala:306]: This method is to set the network tags on the instance:
https://cloud.google.com/compute/docs/reference/rest/v1/instances/setTags

The idea is RBS will add an additional firewall rule in the VPC network which references the new tag which we'll set on Dataproc workers.

Comment (Qi77Qi) [reply google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleComputeInterpreter.scala:306]: ah cool..I thought the method is about the intance

File: google2/CHANGELOG.md [hunk 2]
```diff
@@ -10,12 +10,15 @@ Breaking Changes:
 Changed:
 - Use dataproc client's built in stopCluster method instead of rolling our own
 
+Added:
+- `GoogleComputeService.setInstanceTags`
```
Comment (Qi77Qi) [inline google2/CHANGELOG.md]: maybe add a comment say "for adding network tags"? cuz the method name is a bit misleading

**Review by Qi77Qi — APPROVED** [review]

**Pull request merged**